flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/ExpressionBuilder.java

```diff
@@ -90,9 +91,8 @@ public static UnresolvedCallExpression isNull(Expression input) {
 		return call(IS_NULL, input);
 	}
 
-	public static UnresolvedCallExpression ifThenElse(Expression condition, Expression ifTrue,
-		Expression ifFalse) {
-		return call(IF, condition, ifTrue, ifFalse);
+	public static UnresolvedCallExpression ifThenElse(Expression condition, Expression ifTrue, Expression ifFalse) {
+		return call(IF, call(IS_TRUE, condition), ifTrue, ifFalse);
```

Comment:
Initially I was checking that the first argument was `NOT NULL`. It should not be necessary. I forgot to revert it.

`if the condition is NULL we will always execute the false branch?` <- yes, that will also be the case when I remove it, it will happen down in the code gen.